shared/p2p/service.go

```diff
@@ -215,15 +216,14 @@ func (s *Server) Start() {
 		startPeerWatcher(ctx, s.host, s.bootstrapNode, s.relayNodeAddr)
 	}
 
-	maxTime := time.Duration(1 << 62)
-
-	for _, peer := range s.staticPeers {
-		peerInfo, err := peerInfoFromAddr(peer)
+	for _, staticPeer := range s.staticPeers {
+		peerInfo, err := peerInfoFromAddr(staticPeer)
 		if err != nil {
 			log.Errorf("Invalid peer address: %v", err)
 		} else {
-			s.host.Peerstore().AddAddrs(peerInfo.ID, peerInfo.Addrs, maxTime)
+			s.host.Peerstore().AddAddrs(peerInfo.ID, peerInfo.Addrs, peerstore.PermanentAddrTTL)
 		}
+		startPeerWatcher(ctx, s.host, s.staticPeers...)
```

Comment:
Looks like startPeerWatcher can be called multiple times now. Can we consolidate to one method call? You could build an array of peers to watch and if that list is longer than 0, call startPeerWatcher.